ambry-account/src/main/java/com/github/ambry/account/HelixAccountService.java

```diff
@@ -392,5 +393,22 @@ AccountServiceMetrics getAccountServiceMetrics() {
   BackupFileManager getBackupFileManager() {
     return backupFileManager;
   }
+
+  /**
+   * It gets a collection of {@link Account}s' and {@link Container}s' Id pairs selected by
+   * {@link com.github.ambry.account.Container.ContainerStatus}.
+   */
+  @Override
+  public Set<Pair<Short,Short>> getContainersByStatus(Container.ContainerStatus containerStatus) {
```

Comment:
This method seems like it can be generalized to all implementations of AccountService and implemented as a default method in the interface.

Reply:
+1
I also recommend to return a set of `Container` rather than a set of accountId and containerId.

Reply:
+1

Set of `Container` is friendly.

We can convert this to `Set<Pair<Short,Short>>` in compaction phase to detect DELETE_IN_PROGRESS containers.

Reply:
Updated. And will convert set of containers to Set<Pair<Short,Short>> in compaction phase since we need to check for set membership a lot during compaction stage.